test/framework/autoscaler_helpers.go

```diff
@@ -0,0 +1,276 @@
+	// Note:
+	//  - the file creating the service account to be used by the autoscaler when connecting to the management cluster
+	//    - must be named "autoscaler-to-workload-management.yaml"
+	//    - must deploy objects in the $CLUSTER_NAMESPACE
+	//    - must create a service account with name "cluster-$CLUSTER_NAME" and the RBAC rules required to work.
+	//    - must create a secret with name "cluster-$CLUSTER_NAME-token" and type "kubernetes.io/service-account-token".
+	//  - the file creating the autoscaler deployment in the workload cluster
+	//    - must be named "autoscaler-to-workload-workload.yaml"
+	//    - must deploy objects in the cluster-autoscaler-system namespace
+	//    - must create a deployment named "cluster-autoscaler"
+	//    - must run the autoscaler with --cloud-provider=clusterapi,
+	//      --node-group-auto-discovery=clusterapi:namespace=${CLUSTER_NAMESPACE},clusterName=${CLUSTER_NAME}
+	//      and --cloud-config pointing to a kubeconfig to connect to the management cluster
+	//      using the token above.
+	//    - could use following vars to build the management cluster kubeconfig:
+	//      $MANAGEMENT_CLUSTER_TOKEN, $MANAGEMENT_CLUSTER_CA, $MANAGEMENT_CLUSTER_ADDRESS
+	ArtifactFolder         string
+	InfrastructureProvider string
+	LatestProviderVersion  string
+	AutoscalerVersion      string
+
+	ManagementClusterProxy ClusterProxy
+	Cluster                *clusterv1.Cluster
+	WorkloadClusterProxy   ClusterProxy
+}
+
+// ApplyAutoscalerToWorkloadCluster apply the autoscaler to the workload cluster.
+// Please note that it also create a service account in the workload cluster for the autoscaler to use.
+func ApplyAutoscalerToWorkloadCluster(ctx context.Context, input ApplyAutoscalerToWorkloadClusterInput, intervals ...interface{}) {
+	By("Creating the service account to be used by the autoscaler when connecting to the management cluster")
+
+	managementYamlPath := filepath.Join(input.ArtifactFolder, "repository", fmt.Sprintf("infrastructure-%s", input.InfrastructureProvider), input.LatestProviderVersion, "autoscaler-to-workload-management.yaml")
+	managementYamlTemplate, err := os.ReadFile(managementYamlPath) //nolint:gosec
+	Expect(err).ToNot(HaveOccurred(), "failed to load %s", managementYamlTemplate)
+
+	managementYaml, err := ProcessYAML(&ProcessYAMLInput{
+		Template:             managementYamlTemplate,
+		ClusterctlConfigPath: input.ClusterctlConfigPath,
+		Env: map[string]string{
+			"CLUSTER_NAMESPACE": input.Cluster.Namespace,
+			"CLUSTER_NAME":      input.Cluster.Name,
+		},
+	})
+	Expect(err).ToNot(HaveOccurred(), "failed to parse %s", managementYamlTemplate)
+	Expect(input.ManagementClusterProxy.Apply(ctx, managementYaml)).To(Succeed(), "failed to apply %s", managementYamlTemplate)
+
+	tokenSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: input.Cluster.Namespace,
+			Name:      fmt.Sprintf("cluster-%s-token", input.Cluster.Name),
+		},
+	}
+	Eventually(func() bool {
```

Comment:
slight preference for `Eventually(func(g Gomega) {` and then ending in `.Should(Succeed(),`...

This way we can assert the err and the lookups. E.g.:

`g.Expect(err).NotTo(HaveOccurred())`

and

```go
_, ok := tokenSecret.Data["token"]
g.Expect(ok).To(BeTrue())
```

This way we'll get the line that caused the failure+retry printed to stdout and debugging will be easier.